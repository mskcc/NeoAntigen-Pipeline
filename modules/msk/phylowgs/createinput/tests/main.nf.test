nextflow_process {

    name "Test Process PHYLOWGS_CREATEINPUT"
    script "../main.nf"
    process "PHYLOWGS_CREATEINPUT"

    tag "modules"
    tag "modules_nfcore"
    tag "phylowgs"
    tag "phylowgs/createinput"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("PHYLOWGS_CREATEINPUT - txt") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['cnvs_txt'], checkIfExists: true)
                    ]
                input[1] =  [ 
                    file(params.test_data_mskcc['neoantigen']['unfilteredmaf'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
