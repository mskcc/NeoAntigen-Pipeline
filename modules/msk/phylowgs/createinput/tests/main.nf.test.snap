{
    "PHYLOWGS_CREATEINPUT - txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,9228c0a7bce478e06db5d4304be3dbf7",
                        "ssm_data.txt:md5,055b7b5359a50eefd86fc85fcc023c6f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c3b72d38e738af4cd177ddffac707596"
                ],
                "phylowgsinput": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,9228c0a7bce478e06db5d4304be3dbf7",
                        "ssm_data.txt:md5,055b7b5359a50eefd86fc85fcc023c6f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c3b72d38e738af4cd177ddffac707596"
                ]
            }
        ],
        "timestamp": "2024-03-12T14:20:09.908861"
    }
}